diff --git a/tests/wpst_crm/crm_austin/lib/WPSTUtil.py b/tests/wpst_crm/crm_austin/lib/WPSTUtil.py
index 5c6362d..9f24e5b 100644
--- a/tests/wpst_crm/crm_austin/lib/WPSTUtil.py
+++ b/tests/wpst_crm/crm_austin/lib/WPSTUtil.py
@@ -19,7 +19,7 @@ from oamodule import InterconnectsBase
 
 class WPSTUtil(object):
 
-    def wpst_configure_vc(self, vc_credentials, oa_credentials, config=None, import_domain=True, verbose=True, checkvcmMode=True, doubleDense=False):
+    def wpst_configure_vc(self, vc_credentials, oa_credentials, config=None, import_domain=True, verbose=True, checkvcmMode=True, vcDomainConfigOVBug=False, doubleDense=False):
         try:
             if checkvcmMode is True:
                 vcm_mode_output = self.wpst_execute_oacli_command(oa_credentials, 'show vcmode')
@@ -47,15 +47,20 @@ class WPSTUtil(object):
             """Configuring VC"""
             logger.info("Configuring VC...")
             virtualConnect = VirtualConnect(vc_credentials, oa_credentials)
-            virtualConnect.config_virtual_connect(config, import_domain, verbose, doubleDense)
+            virtualConnect.config_virtual_connect(config, import_domain, verbose, doubleDense, vcDomainConfigOVBug)
 
         except Exception, e:
             logger.info("Attempt to configure VC failed!")
             raise RuntimeError("Attempt to configure VC failed!", e)
 
-    def wpst_delete_domain_vc(self, vc_credentials, oa_credentials):
+    def wpst_delete_domain_vc(self, vc_credentials, oa_credentials, vcDomainConfigOVBug):
         virtualConnect = VirtualConnect(vc_credentials, oa_credentials)
-        virtualConnect.delete_domain()
+        if vcDomainConfigOVBug == 'True':
+            logger.info("Checking if recovery from vcDomainConfigOVBug(QXCR1001433023) is required, please wait...")
+            if not self.wpst_recover_vcm_to_oa_connection_and_destroy_vc_domain_config(vc_credentials, oa_credentials):
+                virtualConnect.delete_domain()
+        else:
+            virtualConnect.delete_domain()
 
     def wpst_clear_vc_mode(self, vc_credentials, oa_credentials):
         virtualConnect = VirtualConnect(vc_credentials, oa_credentials)
@@ -87,6 +92,14 @@ class WPSTUtil(object):
         self.interconnect.execute_diag_cmd(cmd)
         logger.info("%s complete." % cmd)
 
+    def wpst_recover_vcm_to_oa_connection_and_destroy_vc_domain_config(self, vc_credentials, oa_credentials, sleepSec=0):
+        """
+        This is to workaround the defect QXCR1001433023: OV 1.20.06-0207034 VCM Domain config NOT destroyed when moving to OneView
+        """
+        sleep(sleepSec)
+        virtualConnect = VirtualConnect(vc_credentials, oa_credentials)
+        return virtualConnect._vcmOAConnectionRecovery()
+
     def wpst_generate_migrate_task_statistics(self, taskTree, taskGroupSN, taskGroupLN, filename, datafiledir):
         self.wpst_check_dataFiles_dir_exist(datafiledir)
         filename = datafiledir + filename
